File: sl/src/main/java/org/sqlbuilder/sl/PbRoleResolver.java
package org.sqlbuilder.sl;

import org.sqlbuilder.common.DeSerialize;
import org.sqlbuilder.common.Resolver;
import org.sqlbuilder2.ser.Pair;

import java.io.File;
import java.io.IOException;

public class PbRoleResolver extends Resolver<Pair<String, String>, Pair<Integer, Integer>>
{
	public PbRoleResolver(final String ser) throws IOException, ClassNotFoundException
	{
		super(DeSerialize.deserialize(new File(ser)));
	}
}

File: sl/src/main/java/org/sqlbuilder/sl/PbRoleResolver.kt
package org.sqlbuilder.sl

import org.sqlbuilder.common.DeSerialize.deserialize
import org.sqlbuilder.common.Resolver
import java.io.File

typealias SlPbRoleResolvable = Pair<String, String>
typealias SlPbRoleResolved = Pair<Int, Int>

class PbRoleResolver(ser: String) : Resolver<SlPbRoleResolvable, SlPbRoleResolved>(deserialize<Map<SlPbRoleResolvable, SlPbRoleResolved>>(File(ser)))
